apiserver/facades/client/application/application.go

```diff
@@ -1633,6 +1634,21 @@ func (api *APIBase) AddUnits(args params.AddApplicationUnits) (params.AddApplica
 	if api.modelType == state.ModelTypeCAAS {
 		return params.AddApplicationUnitsResults{}, errors.NotSupportedf("adding units on a non-container model")
 	}
+
+	// TODO(wallyworld) - enable-ha is how we add new controllers at the moment
+	// Remove this check before 3.0 when enable-ha is refactored.
+	app, err := api.backend.Application(args.ApplicationName)
+	if err != nil {
+		return params.AddApplicationUnitsResults{}, errors.Trace(err)
+	}
+	ch, _, err := app.Charm()
+	if err != nil {
+		return params.AddApplicationUnitsResults{}, errors.Trace(err)
+	}
+	if ch.Meta().Name == bootstrap.ControllerCharmName {
+		return params.AddApplicationUnitsResults{}, errors.NotSupportedf("add units to the controller application")
```

Comment:
Should this read "add*ing* units ... not supported"?

Reply:
fixed